File: leet-code/subsets/subsets.go
package subsets

func Subsets(nums []int) [][]int {
    n, result := len(nums), [][]int{}
	for i := 0; i < 1 << n; i++ {
		subset := []int{}
		for j := 0; j < n; j++ {
			if i & (1 << j) != 0 {
				subset = append(subset, nums[j])
			}
		}
		result = append(result, subset)
	}
	return result
}

File: leet-code/subsets/subsets.py
from typing import List


class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        result = []
        for i in range(1 << n):
            subset = []
            for j in range(n):
                if i & (1 << j):
                    subset.append(nums[j])
            result.append(subset)
        return result


if __name__ == '__main__':
    obj = Solution()
    obj.subsets([9, 0, 1, 2, 3, 4])
